client/my-sites/plugins/plugins-browser/test/index.jsx

```diff
@@ -82,131 +78,115 @@ const initialReduxState = {
+		render( <PluginsBrowser { ...myProps } /> );
+		expect( screen.queryByText( /no plugins match your search/i ) ).toBeVisible();
 	} );
 	test( 'should show plugin list when there are results', () => {
 		mockPlugins = [ {} ];
-		const comp = mountWithRedux( <PluginsBrowser { ...myProps } /> );
-		expect( comp.find( 'PluginsBrowserList' ).length ).toBe( 1 );
+		render( <PluginsBrowser { ...myProps } /> );
+		expect( screen.queryByText( /found 0 plugins for/i ) ).toBeVisible();
 	} );
 } );
 
 describe( 'Upsell Nudge should get appropriate plan constant', () => {
-	[ PLAN_FREE, PLAN_BLOGGER, PLAN_PERSONAL, PLAN_PREMIUM ].forEach( ( product_slug ) => {
-		test( `Business 1 year for (${ product_slug })`, () => {
-			const comp = mountWithRedux( <PluginsBrowser />, {
-				sites: { items: { 1: { jetpack: false, plan: { product_slug } } } },
-			} );
-			expect(
-				comp.find( 'upsell-nudge[event="calypso_plugins_browser_upgrade_nudge"]' ).length
-			).toBe( 1 );
-			expect(
-				comp.find( 'upsell-nudge[event="calypso_plugins_browser_upgrade_nudge"]' ).props().plan
-			).toBe( PLAN_BUSINESS );
-		} );
-	} );
+	test.each( [ PLAN_FREE, PLAN_BLOGGER, PLAN_PERSONAL, PLAN_PREMIUM ] )(
+		`Business 1 year for (%s)`,
+		( product_slug ) => {
+			const initialState = { sites: { items: { 1: { jetpack: false, plan: { product_slug } } } } };
+			render( <PluginsBrowser />, { initialState } );
+			const nudge = screen.queryByTestId( 'upsell-nudge' );
+			expect( nudge ).toBeVisible();
+			expect( nudge ).toHaveTextContent( PLAN_BUSINESS );
+		}
+	);
 
-	[ PLAN_BLOGGER_2_YEARS, PLAN_PERSONAL_2_YEARS, PLAN_PREMIUM_2_YEARS ].forEach(
+	test.each( [ PLAN_BLOGGER_2_YEARS, PLAN_PERSONAL_2_YEARS, PLAN_PREMIUM_2_YEARS ] )(
+		`Business 2 year for (%s)`,
 		( product_slug ) => {
-			test( `Business 2 year for (${ product_slug })`, () => {
-				const comp = mountWithRedux( <PluginsBrowser />, {
-					sites: { items: { 1: { jetpack: false, plan: { product_slug } } } },
-				} );
-				expect(
-					comp.find( 'upsell-nudge[event="calypso_plugins_browser_upgrade_nudge"]' ).length
-				).toBe( 1 );
-				expect(
-					comp.find( 'upsell-nudge[event="calypso_plugins_browser_upgrade_nudge"]' ).props().plan
-				).toBe( PLAN_BUSINESS_2_YEARS );
-			} );
+			const initialState = {
+				sites: { items: { 1: { jetpack: false, plan: { product_slug } } } },
+			};
+			render( <PluginsBrowser />, { initialState } );
+			const nudge = screen.queryByTestId( 'upsell-nudge' );
+			expect( nudge ).toBeVisible();
+			expect( nudge ).toHaveTextContent( PLAN_BUSINESS_2_YEARS );
 		}
 	);
 } );
 
 describe( 'PluginsBrowser basic tests', () => {
 	test( 'should not blow up and have proper CSS class', () => {
-		const comp = mountWithRedux( <PluginsBrowser /> );
-		expect( comp.find( 'main' ).length ).toBe( 1 );
+		render( <PluginsBrowser /> );
+		const main = screen.queryByRole( 'main' );
+		expect( main ).toBeVisible();
+		expect( main ).toHaveClass( 'main' );
```

Comment:
Seems redundant as it's testing an implementation detail, and an unrelated one at that, since it bleeds into the testing responsibilities of another component.